compiler/src/dotty/tools/dotc/sbt/ExtractAPI.scala

```diff
@@ -445,6 +446,9 @@ private class ExtractAPICollector(implicit val ctx: Context) extends ThunkHolder
         val apiTycon = apiType(tycon)
         val apiArgs = args.map(processArg)
         api.Parameterized.of(apiTycon, apiArgs.toArray)
+      case tp: AppliedTermRef =>
+        val apiTps = (tp.fn :: tp.args).map(apiType)
+        withMarker(combineApiTypes(apiTps: _*), appliedTermRefMarker)
```

Comment:
Could you add an sbt scripted test to check that this is working properly? See https://github.com/lampepfl/dotty/tree/master/sbt-dotty/sbt-test/source-dependencies/by-name for an example.